components/style/build_gecko.rs

```diff
@@ -138,7 +138,12 @@ mod bindings {
             if cfg!(target_os = "linux") {
                 builder = builder.clang_arg("-DOS_LINUX=1");
             } else if cfg!(target_os = "macos") {
-                builder = builder.clang_arg("-DOS_MACOSX=1");
+                builder = builder.clang_arg("-DOS_MACOSX=1")
+                    .clang_arg("-stdlib=libc++")
+                    // To disable the fixup bindgen applies which adds search
+                    // paths from clang command line in order to avoid potential
+                    // conflict with -stdlib=libc++.
+                    .clang_arg("--target=x86_64-apple-darwin");
```

Comment:
I still think creating a bindgen option for skipping it was a better solution, but anyway, this is fine :)